[Issue]: Build fails with -DBUILD_TESTING=OFF

### Problem Description

During building of hipSPARSE and rocSPARSE, if the -DBUILD_TESTING=OFF CMake flag is specified, the install step fails due to missing matrices file.

### Operating System

Ubuntu 24.04.3 LTS (Noble Numbat)

### CPU

AMD RYZEN AI MAX+ PRO 395 w/ Radeon 8060S

### GPU

AMD Radeon 8060S

### ROCm Version

ROCm 7.11.0

### ROCm Component

hipSPARSE

### Steps to Reproduce

cmake -B build -GNinja -DTHEROCK_AMDGPU_FAMILIES=gfx1151 \
  -DCMAKE_C_COMPILER_LAUNCHER=ccache \
  -DCMAKE_CXX_COMPILER_LAUNCHER=ccache \
  -DBUILD_TESTING=OFF \
  .

cmake --build build

### (Optional for Linux users) Output of /opt/rocm/bin/rocminfo --support

_No response_

### Additional Information

_No response_

@ScottTodd fail log as below:
```
FAILED:math-libs/BLAS/rocSPARSE/stamp/stage.stamp /home/nick/Documents/TheRock/TheRock/build/math-libs/BLAS/rocSPARSE/stamp/stage.stamp
cd /home/nick/Documents/TheRock/TheRock/build/math-libs/BLAS/rocSPARSE/build && /home/nick/Documents/TheRock/therock-venv/bin/python3 /home/nick/Documents/TheRock/TheRock/build_tools/teatime.py --log-timestamps --label rocSPARSE\ install --no-interactive /home/nick/Documents/TheRock/TheRock/build/logs/rocSPARSE_install.log -- /usr/bin/cmake --install /home/nick/Documents/TheRock/TheRock/build/math-libs/BLAS/rocSPARSE/build && /home/nick/Documents/TheRock/therock-venv/bin/python3 /home/nick/Documents/TheRock/TheRock/build_tools/fileset_tool.py copy /home/nick/Documents/TheRock/TheRock/build/math-libs/BLAS/rocSPARSE/dist /home/nick/Documents/TheRock/TheRock/build/math-libs/BLAS/rocSPARSE/stage /home/nick/Documents/TheRock/TheRock/build/third-party/sysdeps/linux/zlib/build/stage /home/nick/Documents/TheRock/TheRock/build/third-party/sysdeps/linux/zstd/build/stage /home/nick/Documents/TheRock/TheRock/build/third-party/sysdeps/linux/numactl/build/stage /home/nick/Documents/TheRock/TheRock/build/third-party/sysdeps/linux/bzip2/build/stage /home/nick/Documents/TheRock/TheRock/build/third-party/sysdeps/linux/liblzma/build/stage /home/nick/Documents/TheRock/TheRock/build/third-party/sysdeps/linux/elfutils/build/stage /home/nick/Documents/TheRock/TheRock/build/third-party/sysdeps/linux/libdrm/build/stage /home/nick/Documents/TheRock/TheRock/build/compiler/amd-llvm/stage /home/nick/Documents/TheRock/TheRock/build/compiler/amd-comgr/stage /home/nick/Documents/TheRock/TheRock/build/compiler/amd-comgr-stub/stage /home/nick/Documents/TheRock/TheRock/build/compiler/hipcc/stage /home/nick/Documents/TheRock/TheRock/build/base/rocm-core/stage /home/nick/Documents/TheRock/TheRock/build/base/rocprofiler-register/stage /home/nick/Documents/TheRock/TheRock/build/core/ROCR-Runtime/stage /home/nick/Documents/TheRock/TheRock/build/core/rocminfo/stage /home/nick/Documents/TheRock/TheRock/build/core/clr/stage /home/nick/Documents/TheRock/TheRock/build/profiler/roctracer/stage /home/nick/Documents/TheRock/TheRock/build/profiler/aqlprofile/stage /home/nick/Documents/TheRock/TheRock/build/profiler/rocprof-trace-decoder/stage /home/nick/Documents/TheRock/TheRock/build/third-party/sysdeps/linux/sqlite3/build/stage /home/nick/Documents/TheRock/TheRock/build/profiler/rocprofiler-sdk/stage && /usr/bin/cmake -E touch /home/nick/Documents/TheRock/TheRock/build/math-libs/BLAS/rocSPARSE/stamp/stage.stamp
[rocSPARSE install] -- Install configuration: "Release"
[rocSPARSE install] CMake Error at cmake_install.cmake:46 (file):
[rocSPARSE install]   file INSTALL cannot find
[rocSPARSE install]   "/home/nick/Documents/TheRock/TheRock/build/math-libs/BLAS/rocSPARSE/build/clients/matrices":
[rocSPARSE install]   No such file or directory.
[rocSPARSE install]
[rocSPARSE install]
```

--- a/math-libs/BLAS/pre_hook_hipSPARSE.cmake
+++ b/math-libs/BLAS/pre_hook_hipSPARSE.cmake
@@ -1,6 +1,9 @@
 # See the artifact descriptor where we require these matrices for the test
 # artifact. Consider installing as part of the main project.
-install(
-  DIRECTORY "${CMAKE_CURRENT_BINARY_DIR}/clients/matrices"
-  DESTINATION "clients"
-)
+# The client will only be built if testing is enabled. See hipSPARSE CMakeLists.txt.
+if(THEROCK_BUILD_TESTING)
+  install(
+    DIRECTORY "${CMAKE_CURRENT_BINARY_DIR}/clients/matrices"
+    DESTINATION "clients"
+  )
+endif()

--- a/math-libs/BLAS/pre_hook_rocSPARSE.cmake
+++ b/math-libs/BLAS/pre_hook_rocSPARSE.cmake
@@ -11,7 +11,10 @@ endif()
 
 # See the artifact descriptor where we require these matrices for the test
 # artifact. Consider installing as part of the main project.
-install(
-  DIRECTORY "${CMAKE_CURRENT_BINARY_DIR}/clients/matrices"
-  DESTINATION "clients"
-)
+# The client will only be built if testing is enabled. See rocSPARSE CMakeLists.txt.
+if(THEROCK_BUILD_TESTING)
+  install(
+    DIRECTORY "${CMAKE_CURRENT_BINARY_DIR}/clients/matrices"
+    DESTINATION "clients"
+  )
+endif()

